File: folkers-frontend/src/lib/types/signature.ts
import type { SurrealThing } from "./surreal";

export interface RecordSignatureRecord {
  id: SurrealThing;
  record_id: string;
  base64: string;
  pubkey: string;
  signed_by: string;
}

export interface SignRecordPayload {
  private_key: string;
}

File: folkers-backend/src/database/signature.rs
use surrealdb::sql::Thing;
use serde::{Deserialize, Serialize};

/// Database record with signature content
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RecordSignatureRecord {
    pub id: Option<Thing>,
    pub record_id: String,
    pub base64: String,
    pub pubkey: String,
    pub signed_by: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SignRecordPayload {
    pub private_key: String
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VerifyRecordPayload {
    pub public_key: String
}
